### ISSUE TITLE
Refresh logo for Customizer

### ISSUE THREAD

**Issue by ghost** [kamilion/customizer#115]
**Last revised: 2016.12.19**

This is a long-standing, but trivial issue for Customizer. This issue is created for suggesting to refresh or reproduce existing logo for Customizer at better quality using SVG image. Additionally, a better quality of PNG image may be exported from SVG image.

While this is a trivial issue, I didn't have enough time to reproduce the logo back then, since most of my time were spent on troubleshooting and documentation. Perhaps there is someone else out there to help us reproduce the logo in better quality.

**Justification**

The existing logo is the only image that was designed for Customizer in 2011. The logo file properties: PNG image data, 140 x 140, 8-bit/color RGBA, non-interlaced. Although [PNG](https://en.wikipedia.org/wiki/Portable_Network_Graphics) itself is a lossless graphic file format, the only image available for the logo is low in quality i.e. noticeable pixelated area.

**How to create icon images**

First, create SVG image file at highest quality i.e. 256 x 256 pixels. Then, export to PNG image file at same size i.e. 256 x 256 pixels. Repeat the same for other standard icon sizes. This is the proper method to create icon images, without affecting the image quality (I researched this some time ago).

Regarding the repetitive steps for different icon sizes, I guess it is possible to use a copy of SVG image file at highest quality, then use vector image editor i.e. Inkscape to scale down the SVG image to lower sizes and save as different name for respective sizes.

There will be little to no loss of quality because editing as vector image format, and not bitmap.

**Note** There is a simple maths involved in scaling down the SVG image. To scale down from 256 pixels to 128 pixels, each dimension will be reduced by factor of 128/256=1/2 (50%); From 128 to 48 pixels however, will be reduced by factor of 48/128=0.375 (37.5%). Use these values to calculate the height, width, x and y coordinates of resulting SVG image.

**Source and finished images**

The icon that represents Customizer is most likely derived from ["Tools, Hammer, Spanner" by Andy](https://openclipart.org/detail/67/tools-hammer-spanner) on Openclipart. The source image was finally found in early April 2015, after asking the origin of logo from @fluxer a.k.a the designer.

The finished SVG and PNG images shall be uploaded to master branch, which is the same location as existing logo. The image preview may be linked from there to this Wiki page: [Logos](https://github.com/clearkimura/Customizer/wiki/Logos).

**No installation via `Makefile`**

On the other hand, the icon images at 256, 128, 48 pixels will be kept in a new directory `icons` (This will be created later, prior to future release 4.2.0). The new directory and images will _not_ be included in binary or installation via `Makefile` because these are only useful for users who wants to create custom launcher in dock or larger icon for HiDPI display.

A new pair of SVG and PNG icon images at 24 pixels will be installed via `Makefile`. This is to provide optional icon for authentication dialog. Because of this, other sizes at 256, 128 and 48 pixels could be easily included in `Makefile` in the future, but not necessary for now.

**Above plans are subject to changes and will be decided at last.**

**Comment by fluxer** [comment]
@clearkimura You've missed a dot in the URL for the image.

Edit: never mind, the GMail web interface ate some dots.

**Comment by ghost** [comment]
@fluxer 
This year, almost five years after you had created the original logo, I'd like to share that there will be a new icon image for Customizer. I hope you like the changes made to the logo (see below).

![logo-default-preview](https://cloud.githubusercontent.com/assets/1223304/18287417/a4f963f2-74a9-11e6-9e8c-252599804126.png)

I had used SVG source image and Inkscape to recreate it. Nothing fancy, just a simple, better quality icon image with flat design. I had removed reflective parts and adjusted colour contrast for some objects.

The new icon will be included in future release 4.2.0, which is still under planning.

I have already created the icon images at size of 256, 128, 48 pixels. Just need to schedule my time to commit at later date. Following this, I consider this issue is now "fixed".

**Comment by fluxer** [comment]
@clearkimura 
Glad to see you are still around. And yes, as much I have the word for it I approve the logo - it's flat and clear which is most people seem to like nowdays. Good work! :+1:

**Comment by ghost** [comment]
A quick follow from #148 to provide optional icon.

Using refresh logo for Customizer, I have created another pair of SVG and PNG icon images at size of 24 pixels. This pair of icons will be introduced earlier as part of milestone 4.1.4. Remaining sizes will be introduced later in milestone 4.2.0 as planned.

**TL;DR** `pkexec` implementation varies by desktop environment. The optional icon is just a small part of inconsistencies that I have decided to fix. Other parts such as inconsistent Qt theme has workaround for Qt 4, but that workaround doesn't exist for Qt 5 so far.

**Comment by ghost** [comment]
This issue is considered resolved in last update for "old stable".

Note that `logo.png` still use the original icon image and newly added icon images are *not* installed via Makefile. I have only dump all the SVG and PNG image files in `icons/*` directory for anyone's use.
